prefect_kubernetes/jobs.py

```diff
@@ -0,0 +1,205 @@
+"""Module to define tasks for interacting with Kubernetes jobs."""
+
+from typing import Dict, Optional
+
+from kubernetes import client
+from kubernetes.client.models import V1Job, V1JobList, V1Status
+from prefect import task
+
+from prefect_kubernetes.credentials import KubernetesCredentials
+
+
+@task
+async def create_namespaced_job(
+    body: Dict,
+    kubernetes_credentials: KubernetesCredentials,
+    namespace: Optional[str] = "default",
+    kube_kwargs: Optional[Dict] = None,
+) -> V1Job:
+    """Task for creating a namespaced Kubernetes job.
+
+    Args:
+        body: A dictionary representing a Kubernetes V1Job specification.
+        namespace: The Kubernetes namespace to create this job in.
+        kubernetes_credentials: KubernetesCredentials block
+            holding authentication needed to generate the required API client.
+        kube_kwargs: Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`).
+
+
+    Returns:
+        A Kubernetes `V1Job` object.
+    """
+    api_client = kubernetes_credentials.get_batch_client()
```

Comment:
Double checking whether this needs to be a context managed statement?

Reply:
I don't think so - doesn't appear it can be used as a context manager